### ISSUE TITLE
Could not load type 'HotChocolate.Language.INullabilityNode'

### ISSUE THREAD

**Issue by samyonr** [EntityGraphQL/EntityGraphQL#424]
I'm running `EntityGraphQL.AspNet` version 5.5.0 which internally uses version 13.9.11 of `HotChocolate.Language` and `HotChocolate.Language.SyntaxTree`.

For the client, I'm using `StrawberryShake.Server` version 14.0.0, but I'm getting this error:

```
{"errors":[{"message":"Could not load type \u0027HotChocolate.Language.INullabilityNode\u0027 from assembly \u0027HotChocolate.Language.SyntaxTree, Version=14.0.0.0, Culture=neutral, PublicKeyToken=null\u0027."}]}
```

Note that `StrawberryShake.Server` uses version 14.0.0 of `HotChocolate.Language.SyntaxTree`, which is probably what causes the error.
The error doesn't occur with `StrawberryShake.Server` version 13.9.12.

**Comment by lukemurray** [comment]
Is your client and server in the same project? If they were separate projects (as they shouldn't depend on each other as they talk via GraphQL) this shouldn't be an issue right?

Perhaps I am missing something about your set up.

**Comment by samyonr** [comment]
Not exactly.
You can see my setup in this [gist](https://gist.github.com/samyonr/9fffc791c37d0558ff8e074a6f2f6211).
Or more easily, here: [EntityGraphQL_Hotchocolate14.zip](https://github.com/user-attachments/files/17837050/EntityGraphQL_Hotchocolate14.zip)

In this setup, I have a server that uses EntityGraphQL 5.5.2, a basic package of entities, copied from your [README](https://github.com/EntityGraphQL/EntityGraphQL/blob/master/README.md) (with slight modifications to remove warnings), and two tests projects.

The one named "Tests" uses StrawberryShake.Server 13.9.14 that uses HotChocolate.Language.SyntaxTree 13.9.14 (EntityGraphQL 5.5.2 uses HotChocolate.Language.SyntaxTree version 13.9.11).
The test in "Tests" succeeds.

There's another tests project, named "TestsFail". It's exactly the same project, with the same test and setup, but it uses  StrawberryShake.Server 14.1.0 that uses HotChocolate.Language.SyntaxTree 14.1.0, and the exact same test fails.

My guess is that one of the (breaking changes)[HotChocolate.Language.SyntaxTree ] from HotChocolate.Language.SyntaxTree 13.x.x to 14.x.x causes the issue.

**Comment by lukemurray** [comment]
I see, you are testing the client (I think) but you are bringing in the Server project to create a server/URL to test against. Hence it conflicts.

**Comment by samyonr** [comment]
Well, that's how StrawberryShake [works](https://chillicream.com/docs/strawberryshake/v14/get-started/console):
>Step 3: Install the required packages
>Strawberry Shake supports multiple GraphQL transport protocols. In this example we will use the standard GraphQL over HTTP >protocol to interact with our GraphQL server.
>
>Add the `StrawberryShake.Server` package to your project in order to add our code generation.
>`dotnet add Demo package StrawberryShake.Server`

i.e. to create a client, the only package you need is `StrawberryShake.Server`, which internally uses `HotChocolate.Language.SyntaxTree`.

I don't think that the problem is with client vs server. And there's no conflict in complication. The problem, if I'm not mistaken, is that HotChocolate added some breaking changes moving towards version 14.x.x, and EntityGraphQL doesn't incorporate these changes, yet. That limits the use of EntityGraphQL only with clients that rely on `HotChocolate.Language.SyntaxTree` 13.9.x versions (or 13.x.x).

I didn't try [graphql-client](https://github.com/graphql-dotnet/graphql-client), could be that it works well, but if StrawberryShake is used, which is one of the more popular clients, it prevents from moving forward with the versions.

**Comment by lukemurray** [comment]
I might be missing something as I have never used `StrawberryShake`. But typically EntityGraphQL is used to create the GraphQL Api endpoint. I.e. the server. This is hosted on some web server somewhere.

A client making called to that GraphQL endpoint could be written in any language. It makes calls over HTTP(s).

The issue I _think_ I saw in your project is that the test project includes both the client project (that makes calls over HTTP(s)) and the server project (that using EntityGraphQL to build the end point). I.e. there is no real separation between those projects now.

That all still might be valid for your use case.

I can look at upgraded the dependency, I also have plans to remove it which would solve it for you too.

**Comment by samyonr** [comment]
The setup is indeed as you describe: I have a server that uses GraphQL (EntityGraphQL) and a client, in a separate service, that uses StrawberryShake. 

Note that I use StrawberryShake only in the test (=client) project (and not in the server project), and I'm using EntityGraphQL only in the server project (and not in the test=client project). I do start the server in memory via `CustomWebApplicationFactory` in the test, and reference `..\Server\Server.csproj`, but that's not related - the compilation works well. The problem is in the runtime. It just seems like the latest StrawberryShake client is unable to communicate with EntityGraphQL. If needed, you can start the server separately, create a StrawberryShake client console application, and see that they won't communicate. If you want, I can create a demo project that has it and share it here - let me know. (I hope I don't miss anything... maybe I'll create such a project just in case, to double-check myself).

BTW, in theory, one can have a GraphQL server that is also a GraphQL client of some other server, but that's not the case here.

**Comment by samyonr** [comment]
Actually, I was mistaken.

The GraphQL implementation by EntityGraphQL is either correct, or not, and it doesn't depend on which client is used or which underneath package has been updated. So yes, the problem in my case was that the tests start the server in memory via `CustomWebApplicationFactory` and reference `..\Server\Server.csproj` to do so. The latter, uses version 13.9.11 of `HotChocolate.Language.SyntaxTree`, while the StrawberryShake client in the tests uses version 14.x.x of `HotChocolate.Language.SyntaxTree`.

Once I've created a StrawberryShake client and started the EntityGraphQL server separately, it all worked fine with any of the recent versions of StrawberryShake (that use either `HotChocolate.Language.SyntaxTree` 13.9.11 or 14.x.x).

Sorry for the confusion, and thanks for your responses.
